onadata/apps/api/tests/viewsets/test_user_profile_viewset.py

```diff
@@ -1151,3 +1155,61 @@ def test_monthly_submissions_with_year_param(self):
         self.assertEquals(response.status_code, 200)
         self.assertFalse(self.xform.shared)
         self.assertEquals(response.data, {'private': count})
+
+    def test_change_password_attempts(self):
+        view = UserProfileViewSet.as_view(
+            {'post': 'change_password'})
+
+        # clear cache
+        cache.delete('change_password_attempts-bob')
+        cache.delete('lockout_change_password_user-bob')
+        self.assertIsNone(cache.get('change_password_attempts-bob'))
+        self.assertIsNone(cache.get('lockout_change_password_user-bob'))
+
+        # first attempt
+        current_password = "wrong_pass"
+        new_password = "bobbob1"
+        post_data = {'current_password': current_password,
+                     'new_password': new_password}
+        request = self.factory.post('/', data=post_data, **self.extra)
+        response = view(request, user='bob')
+        self.assertEqual(response.status_code, 400)
+        self.assertEqual(response.data,
+                         "Invalid password."
+                         u" You have 9 attempts left.")
+        self.assertEqual(cache.get('change_password_attempts-bob'), 1)
+
+        # second attempt
+        current_password = "wrong_pass2"
+        new_password = "bobbob1"
+        post_data = {'current_password': current_password,
+                     'new_password': new_password}
+        request = self.factory.post('/', data=post_data, **self.extra)
+        response = view(request, user='bob')
+        self.assertEqual(response.status_code, 400)
+        self.assertEqual(response.data,
+                         "Invalid password. You have 8 attempts left.")
+        self.assertEqual(cache.get('change_password_attempts-bob'), 2)
+
+        # check user is locked out
+        current_password = "wrong_pass"
+        new_password = "bobbob1"
+        post_data = {'current_password': current_password,
+                     'new_password': new_password}
```

Comment:
Lines 1195 to 1198 seem exactly the same as lines 1183 to 1186, I think this is duplicate code that is not necessary.